framework/src/task_manager/task_manager_broadcast.c

```diff
@@ -46,17 +46,20 @@ int task_manager_broadcast(int msg, tm_msg_t *broadcast_data, int timeout)
 	if (request_msg.data == NULL) {
 		return TM_OUT_OF_MEMORY;
 	}
-	if (broadcast_data != NULL) {
+	if (broadcast_data != NULL && broadcast_data->msg_size != 0) {
 		((tm_internal_msg_t *)request_msg.data)->msg = (void *)TM_ALLOC(broadcast_data->msg_size);
 		if (((tm_internal_msg_t *)request_msg.data)->msg == NULL) {
 			TM_FREE(request_msg.data);
 			return TM_OUT_OF_MEMORY;
 		}
 		memcpy(((tm_internal_msg_t *)request_msg.data)->msg, broadcast_data->msg, broadcast_data->msg_size);
 		((tm_internal_msg_t *)request_msg.data)->msg_size = broadcast_data->msg_size;
-	} else {
+	} else if (broadcast_data->msg_size == 0) {
```

Comment:
What happens when broadcast_data == NULL ?

Reply:
If `broadcast_data == NULL`, then it enters into `else` condition (which is line 60).

Reply:
You added additional 'and condition' in line 49.
So you need to add 'broadcast_data != NULL' check in line 57.
Otherwise, a crash may happen.